ste/uploader-blobFS.go

```diff
@@ -169,18 +170,26 @@ func (u *blobFSUploader) GenerateUploadFunc(id common.ChunkID, blockIndex int32,
 	})
 }
 
+var flushThreshold int64 = (1024 ^ 3) * 20 //20 GB flush threshold
+
 func (u *blobFSUploader) Epilogue() {
 	jptm := u.jptm
 
 	// flush
 	if jptm.TransferStatus() > 0 {
 		md5Hash, ok := <-u.md5Channel
+		ss := jptm.Info().SourceSize
 		if ok {
-			//Type assertions aren't my favorite thing to do but it does work when you need it to.
-			_, err := u.fileURL.FlushData(jptm.Context(), jptm.Info().SourceSize, md5Hash, *u.creationTimeHeaders)
-			if err != nil {
-				jptm.FailActiveUpload("Flushing data", err)
-				// don't return, since need cleanup below
+			for i := int64(math.Min(float64(ss), float64(flushThreshold))); i <= ss; i = int64(math.Min(float64(ss), float64(i+flushThreshold))) {
+				_, err := u.fileURL.FlushData(jptm.Context(), i, md5Hash, *u.creationTimeHeaders)
```

Comment:
the md5 hash shouldn't be sent with every request, right? since it's an overall hash of the entire file

Reply:
It's probably not doing any harm, because its put into the x-ms-content-md5 field (which is the whole of file one).   However, there's no benefit in setting it (or the other headers) on any call except the last.